### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by achton**
This PR represents the initial, working version of the action which does sync GH alerts to Jira.

There is no public, functioning implementation of this action, so it is to be viewed from the perspective of someone wanting to implement it in their own repo - so the documentation is also written that way.

Apart from that, things that are relevant to review are:
- unintentional information disclosure (we don't want that)
- robustness of the Symfony console app
- feedback to the dev in the action output (it is very scarce right now, I know)

**Example screenshot** from an automatically created ticket:

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,139 @@
 # github-security-jira
-Github Action for integrating Security Alerts with JIRA
+GitHub Action for mapping security alerts to Jira tickets.
+
+
+## Setup
+
+This action is designed to run in conjunction with another action. In summary, you need the following pieces set up to sync alerts with Jira:
+
+1. Two repo secrets containing a GitHub access token and a Jira API token, respectively.
```
Comment (xendk) [inline README.md]: Actions get a token per default, isn't that sufficient? https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token#about-the-github_token-secret

File: README.md [hunk 2]
```diff
@@ -1,2 +1,139 @@
...
+2. A YAML file which configures the settings for the Jira project where tickets should be created.
+3. A workflow file which runs the two actions on a schedule, continually creating new tickets when necessary.
+
+
+### Jira API Token
+The `reload/github-security-jira` action requires a secret called `JiraApiToken` which should contain an API Token for the Jira user that should be used to create tickets. The token is passed to the action in the `JIRA_TOKEN` environment variable.
+
+
+### Jira config file `gh-jira.yaml`
```
Comment (xendk) [inline README.md]: Maybe it's just me, but I'd prefer if this was configured on the action in the workflow. The minimal configuration is host, user and project, and it seems more suitable in the workflow.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,139 @@
...
+#   Check out the guide [in our wiki|https://foo.atlassian.net/wiki/]!
+```
+
+
+### Workflow file setup
+The [GitHub workflow file](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/configuring-a-workflow#creating-a-workflow-file) should reside in any repo where you want to sync security alerts with Jira.
+
+The workflow file will need a secret called `GitHubSecurityToken` in the repository. This secret should contain a [Personal Access Token](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line) for the GitHub user under which this action should be executed, and the token must include the `public_repo` scope. The token is passed to the action in the `GITHUB_TOKEN` environment variable. Also, the GH user must have access to the repository and [access to security alerts in the repo](https://help.github.com/en/github/managing-security-vulnerabilities/managing-alerts-for-vulnerable-dependencies-in-your-organization).
+
+The workflow uses another GitHub action `octokit/graphql-action` and its output when used to query [GitHub's GraphQL API](https://developer.github.com/v4/).
```
Comment (xendk) [inline README.md]: Isn't there a Symfony GraphQL package? Having to configure a different action just to get the data (with the danger of messing that up) seems a bit overwhelming.

File: bin/ghsec-jira [hunk 4]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env php
+<?php
+// The following way to require composers autoload.php is borrowed
+// from https://github.com/sebastianbergmann/phpunit
+foreach ([__DIR__ . '/../../autoload.php', __DIR__ . '/../vendor/autoload.php', __DIR__ . '/vendor/autoload.php'] as $file) {
```
Comment (xendk) [inline bin/ghsec-jira]: The third option is not really needed. First covers installing this project in another project (which I guess covers global installs too), but is that really needed? Second case covers cloning the project and doing a `composer install`. And the last? I really don't know when you'd have that case.

File: composer.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        "lesstif/php-jira-rest-client": "^1",
+        "symfony/yaml": "^5.0"
+    },
+    "autoload": {
+        "psr-4": {
+            "GitHubSecurityJira\\": "src/"
+        }
+    },
+    "bin": [
+        "ghsec-jira"
```
Comment (xendk) [inline composer.json]: This should be `bin/ghsec-jira`. But it only matters when this project is installed as a dependency of another, in which case composer links the files into `vendor/bin`.

File: composer.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    ],
+    "require-dev": {
+        "squizlabs/php_codesniffer": "^3.0"
```
Comment (xendk) [inline composer.json]: Maybe you'd be interested in https://github.com/appocular/coding-standard ? Much more opinionated than PSR12, but I like it (I ought to, don't I?).

File: src/JiraIssue.php [hunk 7]
```diff
@@ -0,0 +1,260 @@
+<?php
+
```
Comment (xendk) [inline src/JiraIssue.php]: No `declare(strict_types=1);`? :smiley:

File: README.md (hunk 1 above)
Comment (achton) [reply README.md]: AFAICT the default token does not have access to security alerts.

File: README.md (hunk 2 above)
Comment (achton) [reply README.md]: Yes, that's probably a better fit actually.

File: README.md (hunk 3 above)
Comment (achton) [reply README.md]: I agree. With the change above, a minimum workflow would become:
```yaml
jobs:
  syncSecurityAlerts:
    name: "Get security alerts for this repo"
    runs-on: ubuntu-latest
    steps:
      - name: "Sync alert data to Jira tickets"
        uses: reload/github-security-jira@v1.x
        env:
          GITHUB_TOKEN: ${{ secrets.GitHubSecurityToken }}
          JIRA_TOKEN: ${{ secrets.JiraApiToken }}
          JIRA_HOST: <hostname>
          JIRA_USER: <username>
          JIRA_PROJECT: <project id>
```

in any repo that wants to use this action.

Agreed?

Comment (arnested) [reply README.md]: I concur!

File: bin/ghsec-jira (hunk 4 above)
Comment (arnested) [reply bin/ghsec-jira]: I'm pretty sure @achton has it from the code I wrote for https://github.com/reload/lumturio-jira. And I stole it from phpunit (as the comment says). I'm not sure / don't remember what use cases it's supposed to cover. I think one of them also covers putting all the stuff into a PHAR file.

File: README.md (hunk 3 above)
Comment (xendk) [reply README.md]: @achton 
Totally

File: bin/ghsec-jira (hunk 4 above)
Comment (xendk) [reply bin/ghsec-jira]: AFAIK, a phar is just an archive, so relative paths work the same.

Comment (arnested) [reply bin/ghsec-jira]: Ok, dunno then. Except phpunit does it :)

Comment (xendk) [reply bin/ghsec-jira]: PHPUnit uses an XML config file. I rest my case.

File: composer.json (hunk 6 above)
Comment (achton) [reply composer.json]: I went with straight PSR-12 for now :)

**Pull request merged**